### PR TITLE
Addresses issue 122

### PR DESCRIPTION
**Pull request by wwwtyro**
See https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/122

### CODE REVIEW

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 1]
```diff
@@ -197,10 +195,12 @@ def render_and_launch_ingress(kube_dns):
 def scale_ingress_controller():
     ''' Scale the number of ingress controller replicas to match the number of
     nodes. '''
-    output = kubectl('get', 'nodes', '-o', 'name')
-    count = len(output.splitlines())
-    if data_changed('ingress-controller-replicas', count):
```
Comment (wwwtyro) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: In testing, running `kubectl scale` where the scale doesn't change appears to be a noop, so removing this check.

**Comment by Cynerva** [comment]
Tested a fresh deployment with two workers, hook failure no longer occurs:
```
$ juju show-status-log kubernetes-worker/13
TIME                 	TYPE     	STATUS   	MESSAGE                                   
04 Nov 2016 15:40:10Z	juju-unit	idle     	                                          
04 Nov 2016 15:40:29Z	juju-unit	executing	running kube-dns-relation-changed hook    
04 Nov 2016 15:40:31Z	workload 	waiting  	Waiting for kubelet to start.             
04 Nov 2016 15:40:31Z	juju-unit	idle     	                                          
04 Nov 2016 15:44:40Z	juju-unit	executing	running certificates-relation-joined hook 
04 Nov 2016 15:44:40Z	workload 	waiting  	Waiting for kubelet to start.             
04 Nov 2016 15:44:41Z	juju-unit	executing	running certificates-relation-changed hook
04 Nov 2016 15:44:45Z	workload 	waiting  	Waiting for kubelet to start.             
04 Nov 2016 15:44:47Z	workload 	active   	Kubernetes worker running.                
04 Nov 2016 15:44:48Z	juju-unit	idle     	                                          
04 Nov 2016 15:49:40Z	juju-unit	executing	running update-status hook                
04 Nov 2016 15:49:42Z	workload 	active   	Kubernetes worker running.                
04 Nov 2016 16:02:09Z	         	idle     	last 1 statuses repeated 1 times          
04 Nov 2016 15:49:44Z	juju-unit	idle     	                                          
04 Nov 2016 15:54:40Z	juju-unit	executing	running update-status hook                
04 Nov 2016 15:54:40Z	workload 	active   	Kubernetes worker running.                
04 Nov 2016 16:02:09Z	         	idle     	last 3 statuses repeated 1 times
```

Ingress is still deployed and scaled correctly:
```
$ kubectl get po
NAME                             READY     STATUS    RESTARTS   AGE
default-http-backend-ypmid       1/1       Running   0          19m
nginx-ingress-controller-1riuy   1/1       Running   0          19m
nginx-ingress-controller-arbqj   1/1       Running   0          19m
```

Changes look fine to me. @chuckbutler @mbruzek could one or both of you have a look as well? Thanks!

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 2]
```diff
@@ -178,11 +178,9 @@ def render_and_launch_ingress(kube_dns):
     ''' If configuration has ingress RC enabled, launch the ingress load
     balancer and default http backend. Otherwise attempt deletion. '''
     config = hookenv.config()
-    # If ingress is enabled, launch the ingress controller and open ports
+    # If ingress is enabled, launch the ingress controller
     if config.get('ingress'):
         launch_default_ingress_controller()
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: I'm unsure why the ports were removed here...

File: cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py [hunk 3]
```diff
@@ -311,9 +311,15 @@ def launch_default_ingress_controller():
     # Render the ingress replication controller manifest
     manifest = addon_path.format('ingress-replication-controller.yaml')
     render('ingress-replication-controller.yaml', manifest, context)
-    kubectl_manifest('create', manifest)
-    hookenv.log('Creating the ingress replication controller.')
-    set_state('kubernetes-worker.ingress.available')
+    if kubectl_manifest('create', manifest):
+        hookenv.log('Creating the ingress replication controller.')
+        set_state('kubernetes-worker.ingress.available')
+        hookenv.open_port(80)
```
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py]: ok, it makes more sense to have that here when we're taking an action on the ingress LB. Be explicit about opening those ports. +1 THIS LGTM

**Pull request merged**